core/common/src/main/scala/org/typelevel/otel4s/AttributeKey.scala

```diff
@@ -80,4 +80,7 @@ object AttributeKey {
   implicit def attributeKeyShow[A]: Show[AttributeKey[A]] =
     Show.show(key => show"${key.`type`}(${key.name})")
 
+  implicit val attributeKeyAnyHash: Hash[AttributeKey[_]] =
+    Hash.by(key => (key.name, key.`type`))
```

Comment:
This is a bit weird, we definitely don't want to keep it gated in a testkit or behind an extra import?

Either way, `Any` is distinct from `_` which I think is called "existential".

Reply:
I guess we can keep it behind the certain import. 

There are several places in (upcoming) Otel4s SDK where we don't know a concrete type of the Attritube

Reply:
Could you suggest the object's name where I can place this implicit, please? 

Should the structure be similar to something like this?:

```scala
object AttributeKey {
  object Implicits {
    object Extra {
      implicit val attributeKeyExistentialHash: Hash[AttributeKey[_]] = ...
    }
  }
}
```

Reply:
Hm, we already define some `Show[AttributeType[_]]` instances in the public implicit lookup scope: https://github.com/typelevel/otel4s/blob/571900a62b0d55c7ad529a14b43325fe7437341f/core/common/src/main/scala/org/typelevel/otel4s/Attribute.scala#L73.

Reply:
@armanbilge what are the implications if we keep existential instances in the public scope? 

We need existential instances for:
1) `AttributeType[_]` - a sealed trait with case objects, universal hashcode is fine here
2) `AttributeKey[_]` - created from `key => (key.name, key.type)`, where the `name` is a `String` and the `type` is an `AttributeType[_]` (a coproduct)
3) `Attribute[_]` - created from `a => (a.key, a.value)`, where the `key` is `AttributeKey`, the `value` is one of Boolean, Double, String, Long, List[Boolean], List[Double], List[String], List[Long].

Reply:
If we are confident that universal hashing will work for all the possible types (now and future) then it seems fine. e.g. could there ever be an attribute for a `CIString`?

Reply:
[Spec](https://opentelemetry.io/docs/specs/otel/common/) says the following:

> The attribute value is either:
> - A primitive type: string, boolean, double precision floating point (IEEE 754-1985) or signed 64 bit integer.
> - An array of primitive type values. The array MUST be homogeneous, i.e., it MUST NOT contain values of different types.

I don't think we should allow any other types other than these.
So, I doubt `Attribute[CIString]` should be a thing.

Reply:
> regardless, `CIString` has a case-insensitive `hashCode` implementation

Yes, it does today, but I'm considering a future where we might want to make it an `opaque type`.